src/Eloquent/Concerns/InteractsWithRelations.php

```diff
@@ -94,9 +94,17 @@ protected function fillConnections(array $connections)
      * @param Relations\BelongsTo $relation
      * @param mixed $id
      * @return void
+     * @throws \Illuminate\Auth\Access\AuthorizationException
      */
     protected function connectBelongsToRelation(Relations\BelongsTo $relation, $id)
     {
+        $current = $relation->first();
+
+        if ($current) {
```

Comment:
Does this mean we always call authorizeToRemove when there’s an existing relation? Seems a bit overkill if you are just updating it.

Reply:
What do you mean with `just updating it`?

```
$old = [
  'related_id' => '1',
]
$input = [
  'related_id' => '1',
]
```

I do however believe that when you're not actually _changing_ the relation, it should not need to authorize changing the fields. (e.g. only check dirty/changed fields)

Reply:
If you would update it with `'related_id' => '2'` it would trigger the `authorizeToRemove` that feels unnecessary. Definitely agree on skipping authorisation when it's not dirty.